### ISSUE TITLE
Support For AngularNodeAppEngine

### ISSUE THREAD

**Issue by Char2sGu** [stevermeister/ngx-cookie-service#335]
### Is your feature request related to a problem? Please describe

The new AngularNodeAppEngine in developer preview enables hybrid rendering, and also provides the request/response context to build-in tokens from the `@angular/core` package. However, with `AngularNodeAppEngine`, we are no longer allowed to configure custom providers in `servrer.ts`.

### Describe the solution you'd like

Should try to fetch the request information from the built-in `REQUEST` token:

```ts
import { REQUEST } from '@angular/core';
import { REQUEST as NGX_COOKIE_REQUEST } from 'ngx-cookie-service-ssr';
```

```ts
    {
      provide: NGX_COOKIE_REQUEST,
      useFactory: (request = inject(REQUEST)) => ({
        headers: {
          cookie: request?.headers.get('cookie') ?? '',
        },
      }),
    },
```

### Describe alternatives you've considered

Document the usage with AngularNodeAppEngine.

**Comment by bballardkele** [comment]
> Should try to fetch the request information from the built-in REQUEST token

I agree with the proposed solution. The package should probably pivot to using the tokens provided by the `@angular/core` package instead of exposing shared tokens from `@ngx-cookie-service-ssr`. It's interesting that the Angular team removed these tokens as a part of the rebranding/retirement of Universal, just to bring them back in the v19 preview.

Also see the `RESPONSE_INIT` token which if I'm understanding correctly will need to replace the `@ngx-cookie-service-ssr RESPONSE` token introduced in #320 . https://angular.dev/guide/hybrid-rendering#accessing-request-and-response-via-di
